csharp/src/Google.Protobuf/JsonParser.cs

```diff
@@ -67,15 +67,18 @@ public sealed class JsonParser
 
         private static readonly JsonParser defaultInstance = new JsonParser(Settings.Default);
 
-        private static readonly Dictionary<string, Action<JsonParser, IMessage, JsonTokenizer>>
-            WellKnownTypeHandlers = new Dictionary<string, Action<JsonParser, IMessage, JsonTokenizer>>
+        // TODO: Consider introducing a class containing parse state of the parser, tokenizer and depth. That would simplify these handlers
```

Comment:
Yes this sounds like a pretty good idea, but I agree there's no need to do it in this PR.

Btw, it looks like that one place in the code where keeping track of recursion depth would be very straigtforward is JsonTokenizer - because tokenizer sees the beginning and end of json objects.
In a way, this might be more accurate because some well known types (e.g. struct) will be counted in terms of their real parsing depth - but I am not sure this is actuall the behavior we want.